component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+                                "last_modified string, status string);\n" +
+                                "@sink(type='log')" +
+                                "define stream FooStream (filepath string, length int, " +
+                                "last_modified string, status string); \n",
+
+                        description = "" +
+                                "Under above configuration, all the files in the specific uri \n" +
+                                "which has been  modified or created or deleted in the execution time " +
+                                "will be identified and displayed. \n After monitoring has been started," +
+                                " If any files in the specific uri is newly created, modified or deleted.\n " +
+                                "an event is created with the filepath, length, last modified date " +
+                                "and status of the file. Then that will be received by the FooStream. If there " +
+                                "are any changes a new event will be generated in every 200 milliseconds.\n"
+                ),
+
+        }
+)
+
+public class FileHandler extends Source<FileHandler.FileHandlerState> {
+    private static final Logger log = Logger.getLogger(FileHandler.class);
+    private SourceEventListener sourceEventListener;
+    private long monitoringInterval = 1000;
+    private String listeningUri;
+    private String listeningDirUri;
+    private String listeningFileUri;
+    private FileAlterationMonitor monitor;
+    private Map<String, Long> initialMap = new ConcurrentHashMap<>();
+    private Map<String, Long> recentFileSizeMap = new ConcurrentHashMap<>();
+    private SiddhiAppContext siddhiAppContext;
+    private List<ThreadOnContinuousChange> threadList = new ArrayList<>();
+    private static final String CURRENT_MAP_KEY = "current.map.key";
+    private static final String RECENT_FILE_SIZE_MAP_KEY = "recent.file.size.map.key";
+
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public StateFactory<FileHandlerState> init(SourceEventListener sourceEventListener,
+                                               OptionHolder optionHolder,
+                                               String[] requiredProperties,
+                                               ConfigReader configReader,
+                                               SiddhiAppContext siddhiAppContext) throws SiddhiAppValidationException {
+        this.siddhiAppContext = siddhiAppContext;
+        this.sourceEventListener = sourceEventListener;
+        if (optionHolder.isOptionExists(Constants.URI)) {
+            listeningUri = optionHolder.validateAndGetStaticValue(Constants.URI);
+        }
+        if (listeningUri == null) {
+            throw new SiddhiAppCreationException(" uri must be provided.");
```

Comment:
Also, check for listeningUri.isEmpty() as well.
Exception message starting space has to be removed, and the sentence should be started with capital.

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/3f9a2a0eb50655821dd322688aa034be33ef8860